src/remoteCollection/handleCollectResponse.js

```diff
@@ -114,17 +115,13 @@ function handleCollectResponse(collectResponse) {
        * error samples. Use the first one to match. If 200 is matched, it
        * will override the default transform.
        */
-      if (tr.errorHandlers) {
-        Object.keys(tr.errorHandlers).forEach((statusMatcher) => {
-          const re = new RegExp(statusMatcher);
-          if (re.test(status)) {
-            func = tr.errorHandlers[statusMatcher];
-          }
-        });
+      const matcherFunc = RefocusCollectorEval.statusCodeMatch(tr, status);
+      if (matcherFunc) { // undefined if no status match
+        func = matcherFunc;
```

Comment:
why not just do:
const func = RefocusCollectorEval.statusCodeMatch(tr, status)

Reply:
Also in statusCodeMatch you can just do:
"return tr.errorHandlers[statusMatcher];"
and get rid of "let func;" and "return func;"

Reply:
func variable is already defined, we cannot reuse it as "const func".

So, lets say I do "return tr.errorHandlers[statusMatcher];" inside the if condition. And we do "func = RefocusCollectorEval.statusCodeMatch(tr, status);"
func will be reassigned undefined when the if condition is not met. I don't think this was expected of original code. We were reassigning func only when the if condition is met. Makes sense? Am I missing something?

Reply:
You're right. It's good as-is.

Reply:
My second comment would still be a good change though, I think. If it doesn't match anything it will still return undefined.

Reply:
Right, I will make that change.

Reply:
@jgraff2 Moving statusCodeMatch to the refocus-collector-eval makes 6 of handleCollectResponse.js handler match tests fail. I tried to investigate, but looks like I would need to spend more time to figure out why. Do you have any idea why they might be failing? If it is not straightforward, we should probably have a story on moving statusCodeMatch to refocus-collector-eval altogether, given we need to add independent tests for same in new repo.

Reply:
@iamigo @jgraff2 Ok if I create a new story for this and keep the statusCodeMatch code here?

Reply:
That's fine.
Looks like the tests are failing because handleCollectResponse is returning default samples. This means statusCodeMatch() is returning undefined.

Reply:
Which is not expected. Though code looks fine in refocus-collector-eval. I will create a story and update this PR. Thanks!